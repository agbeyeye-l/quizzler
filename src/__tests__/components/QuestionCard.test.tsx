import React from "react";
import { render } from "@testing-library/react";
import QuestionCard from "../../components/QuestionCard";

const userAnswer = {
  question:
    "Half-Life by Valve uses the GoldSrc game engine, which is a highly modified version of what engine?",
  correct: true,
  correctAnswer: "Quake Engine",
  answer: "Quake Engine",
};

const props = {
  userAnswer,
  question:
    "Half-Life by Valve uses the GoldSrc game engine, which is a highly modified version of what engine?",
  answers: ["Doom Engine", "id Engine", "Source Engine", "Quake Engine"],
  callback: jest.fn(),
};

// Basic tests, this doesn't exhaust all that can be tested for this application.
// More would be added as time goes on.
describe("QuestionCard Tests", () => {
  it("renders", () => {
    expect(
      render(
        <QuestionCard
          question={props.question}
          answers={props.answers}
          callback={props.callback}
          questionNo={0}
          totalQuestion={10}
          userAnswer={props.userAnswer}
        />
      )
    ).toMatchSnapshot();
  });
});
