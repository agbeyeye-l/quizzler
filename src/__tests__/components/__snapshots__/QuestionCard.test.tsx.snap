// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard Tests renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM kURkQH"
      >
        <p
          class="number"
        >
          Question: 
          0
           / 
          10
        </p>
        <p>
          Half-Life by Valve uses the GoldSrc game engine, which is a highly modified version of what engine?
        </p>
        <div>
          <div
            class="sc-gtsrHT lnMfvG"
          >
            <button
              data-testid="optionsButton"
              disabled=""
              value="Doom Engine"
            >
              <span>
                Doom Engine
              </span>
            </button>
          </div>
          <div
            class="sc-gtsrHT lnMfvG"
          >
            <button
              data-testid="optionsButton"
              disabled=""
              value="id Engine"
            >
              <span>
                id Engine
              </span>
            </button>
          </div>
          <div
            class="sc-gtsrHT lnMfvG"
          >
            <button
              data-testid="optionsButton"
              disabled=""
              value="Source Engine"
            >
              <span>
                Source Engine
              </span>
            </button>
          </div>
          <div
            class="sc-gtsrHT fVtpvm"
          >
            <button
              data-testid="optionsButton"
              disabled=""
              value="Quake Engine"
            >
              <span>
                Quake Engine
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM kURkQH"
    >
      <p
        class="number"
      >
        Question: 
        0
         / 
        10
      </p>
      <p>
        Half-Life by Valve uses the GoldSrc game engine, which is a highly modified version of what engine?
      </p>
      <div>
        <div
          class="sc-gtsrHT lnMfvG"
        >
          <button
            data-testid="optionsButton"
            disabled=""
            value="Doom Engine"
          >
            <span>
              Doom Engine
            </span>
          </button>
        </div>
        <div
          class="sc-gtsrHT lnMfvG"
        >
          <button
            data-testid="optionsButton"
            disabled=""
            value="id Engine"
          >
            <span>
              id Engine
            </span>
          </button>
        </div>
        <div
          class="sc-gtsrHT lnMfvG"
        >
          <button
            data-testid="optionsButton"
            disabled=""
            value="Source Engine"
          >
            <span>
              Source Engine
            </span>
          </button>
        </div>
        <div
          class="sc-gtsrHT fVtpvm"
        >
          <button
            data-testid="optionsButton"
            disabled=""
            value="Quake Engine"
          >
            <span>
              Quake Engine
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
